### PR TITLE
chart: support Rancher deployed Windows Cluster

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/4246

### CODE REVIEW

File: chart/templates/default-setting.yaml [hunk 1]
```diff
@@ -20,8 +20,16 @@ data:
...
     {{ if not (kindIs "invalid" .Values.defaultSettings.backupstorePollInterval) }}backupstore-poll-interval: {{ .Values.defaultSettings.backupstorePollInterval }}{{ end }}
-    {{ if not (kindIs "invalid" .Values.defaultSettings.taintToleration) }}taint-toleration: {{ .Values.defaultSettings.taintToleration }}{{ end }}
-    {{ if not (kindIs "invalid" .Values.defaultSettings.systemManagedComponentsNodeSelector) }}system-managed-components-node-selector: {{ .Values.defaultSettings.systemManagedComponentsNodeSelector }}{{ end }}
+    {{- if or (not (kindIs "invalid" .Values.defaultSettings.taintToleration)) (not (kindIs "invalid" .Values.global.cattle.windowsCluster.enabled)) }}
+    taint-toleration: {{$windowsDefaultSettingTaintToleration := list -}}
+      {{- if .Values.global.cattle.windowsCluster.enabled }}{{ $windowsDefaultSettingTaintToleration = .Values.global.cattle.windowsCluster.defaultSetting.taintToleration }}{{ end -}}
+      {{- $taintToleration := list $windowsDefaultSettingTaintToleration .Values.defaultSettings.taintToleration }}{{ join ";" (compact $taintToleration) -}}
```
Comment (shuo-wu) [inline chart/templates/default-setting.yaml:26]: Should we check `(not (kindIs "invalid" .Values.defaultSettings.taintToleration))` before applying this?

Comment (c3y1huang) [reply chart/templates/default-setting.yaml:28]: Updated, thanks!

Comment (c3y1huang) [reply chart/templates/default-setting.yaml]: Updated, thanks!

**Review by shuo-wu — APPROVED** [review]

File: chart/values.yaml [hunk 2]
```diff
@@ -4,6 +4,22 @@
 global:
   cattle:
     systemDefaultRegistry: ""
+    windowsCluster:
+      # Enable this to allow Longhorn to run on the Rancher deployed Windows cluster
+      enabled: false
```
[the remark this answers is not in the document]
Comment (innobead) [reply chart/values.yaml:9]: After discussing with @c3y1huang , we agree that adding the vendor-specific annotations automatically is not suitable here, so we will just keep the current way as is first, then see the feedback.

**Review by innobead — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
@c3y1huang Please backport to 1.3.x and 1.2.x. Thanks.

**Pull request merged**